e2e/cluster/docker/cluster.go

```diff
@@ -28,13 +28,24 @@ type ClusterInput struct {
 
 func NewCluster(in *ClusterInput) *Cluster {
 	c := &Cluster{t: in.T}
-	for i := 0; i < in.Nodes; i++ {
-		node := NewNode(in, fmt.Sprintf("node%d", i))
-		if i == 0 {
-			node = node.WithPort("30003:30003")
-		}
-		c.Nodes = append(c.Nodes, node)
+
+	c.Nodes = make([]*Container, in.Nodes)
+
+	wg := sync.WaitGroup{}
+	wg.Add(in.Nodes)
+
+	for i := range in.Nodes {
+		go func(i int) {
+			defer wg.Done()
+			node := NewNode(in, fmt.Sprintf("node%d", i))
+			if i == 0 {
+				node = node.WithPort("30003:30003")
+			}
+			c.Nodes[i] = node
+		}(i)
 	}
+	wg.Wait()
```

Comment:
I don't think NewNode runs anything. You'll probably need to make your changes in the "Run" function below as that's what starts the containers.